### PR TITLE
statistics: fix three bugs about MergePartTopN2GlobalTopN

### LINKED ISSUE #24349: some bugs about `MergePartTopN2GlobalTopN`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

When we test the accuracy of global stats， we found that TiDB entered an endless loop. Here is the debug stack
image

Through this, we found a bug in the function `func (hg *Histogram) RemoveVals(valCntPairs []TopNMeta)`： If the `valCntPairs` has a value which is less than lower bound of hg.buckets[0], it will enter an endless loop.

Then we try to write a test case to trigger this bug. But in the process, we found two bugs in `MergePartTopN2GlobalTopN`:

1. It uses incorrect encode value to build `TopNMeta` struct.
2. `func (hg *Histogram) RemoveVals(valCntPairs []TopNMeta)` needs `valCntPairs` in ascending order， but `MergePartTopN2GlobalTopN` doesn't guarantee this assumption.

Here is another issue about `analyze version`.
For version1, histograms contain `topn` values, but version2 histograms do not.
Considering that different versions have greater complexity, and if `topn` is not included in the histogram, the stats will be more accurate. For global stats, we will uniformly remove the `topn` value in the histogram.

It is difficult to provide integration test scenarios for these issues, I will provide unit tests in a later PR.

### PR DESCRIPTION
**Pull request by rebelice**
### What problem does this PR solve?

Issue Number: close #24349 

Problem Summary:

Please check the issue for details

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 

- No release note

### CODE REVIEW

File: statistics/histogram.go [hunk 1]
```diff
@@ -352,7 +353,8 @@ func (hg *Histogram) RemoveVals(valCntPairs []TopNMeta) {
 			// If the current val smaller than current bucket's lower bound, skip it.
 			cmpResult = bytes.Compare(hg.Bounds.Column(0).GetRaw(bktIdx*2), valCntPairs[pairIdx].Encoded)
 			if cmpResult > 0 {
-				continue
+				pairIdx++
```
Comment (qw4990) [inline statistics/histogram.go:356]: Should we increase `pairIdx` also at line:362?

Comment (winoros) [reply statistics/histogram.go:356]: For line 362, it's the case `bucket.upper < current pair`. We need to go to the next bucket. The pair doesn't need to move.

File: statistics/histogram.go [hunk 2]
```diff
@@ -352,7 +353,8 @@ func (hg *Histogram) RemoveVals(valCntPairs []TopNMeta) {
...
+				break
```
Comment (winoros) [inline statistics/histogram.go:357]: It should be `continue`. We move to next pair. The bucket doesn't need to move currently.

File: statistics/histogram.go (hunk 1 above)
Comment (rebelice) [reply statistics/histogram.go:356]: No. If the current val bigger than the current bucket's upper bound, we need to increase `bktIdx`. So we only break here.

File: statistics/histogram.go (hunk 2 above)
Comment (rebelice) [reply statistics/histogram.go:357]: oh, my mistake...

Comment (rebelice) [reply statistics/histogram.go]: updated.

**Review by qw4990 — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**